cmd/object_ls.go

```diff
@@ -0,0 +1,101 @@
+/***************************************************************
+*
+* Copyright (C) 2024, Pelican Project, Morgridge Institute for Research
+*
+* Licensed under the Apache License, Version 2.0 (the "License"); you
+* may not use this file except in compliance with the License.  You may
+* obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+***************************************************************/
+
+package main
+
+import (
+	"os"
+
+	"github.com/pelicanplatform/pelican/client"
+	"github.com/pelicanplatform/pelican/config"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var (
+	lsCmd = &cobra.Command{
+		Use:   "ls {object}",
+		Short: "List objects in a namespace from a federation",
+		Run:   listMain,
+	}
+)
+
+func init() {
+	flagSet := lsCmd.Flags()
+	flagSet.StringP("token", "t", "", "Token file to use for transfer")
+	flagSet.BoolP("long", "L", false, "Include extended information")
+	flagSet.BoolP("dironly", "D", false, "List directories only")
+	flagSet.BoolP("fileonly", "F", false, "List files only")
```

Comment:
I'll point out that I worry about referring to things as "files" when the second sentence of our own client documentation states:
```
One thing to note is that Pelican should be thought of as a tool that works with federated objects as opposed to files.
```
Can we change whichever is incorrect?